### PR TITLE
Regression tests: Token Type

### LINKED ISSUE #2914: Add comparison tests for Token Type validation in new validation model
Need to add tests to compare the new validation path against the existing one to ensure no regressions are introduced in the new path and that all scenarios behave as expected.

- Add tests comparing `ValidateTokenAsync` in the new path against the existing one for Token Type scenarios.

Part of #2711

### PR DESCRIPTION
**Pull request by iNinja**
# Regression tests: Token Type
- Added TokenTypeValidationError to create the exception and assign the invalid token type.
- Renamed ValidationParameters.TypeValidator to ValidationParameters.TokenTypeValidator
- Adjusted the creation of SecurityTokenInvalidTypeException to return the correct invalid type when it is empty or null
- Added regression/comparison tests for token type validation scenarios

Resolves #2914 

Part of #2711

### CODE REVIEW

File: src/Microsoft.IdentityModel.Tokens/Validation/ValidationParameters.cs [hunk 1]
```diff
@@ -499,7 +499,7 @@ public TokenReplayValidationDelegate TokenReplayValidator
         /// </remarks>
         /// <exception cref="ArgumentNullException">Thrown when the value is set as null.</exception>
         /// <returns>The <see cref="TokenTypeValidationDelegate"/> used to validate the token type of a token</returns>
-        public TokenTypeValidationDelegate TypeValidator
+        public TokenTypeValidationDelegate TokenTypeValidator
```
Comment (kllysng) [inline src/Microsoft.IdentityModel.Tokens/Validation/ValidationParameters.cs:502]: Is this a breaking change?

**Review by kllysng — CHANGES_REQUESTED** [review]
Please see my comment regarding the breaking change in ValidationParameters

File: src/Microsoft.IdentityModel.Tokens/Validation/ValidationParameters.cs (hunk 1 above)
Comment (iNinja) [reply src/Microsoft.IdentityModel.Tokens/Validation/ValidationParameters.cs:502]: The type `ValidationParameters` is internal and not used anywhere outside of our new `ValidateTokenAsync`.
So I would say not a public interface change, even if the property is marked as public.

Comment (kllysng) [reply src/Microsoft.IdentityModel.Tokens/Validation/ValidationParameters.cs:502]: That's right, thanks Ignacio. Resolving this thread!

**Review by kllysng — APPROVED** [review]
Thank you @iNinja

**Review by westin-m — APPROVED** [review]

**Pull request merged**